lib/build-bundler.js

```diff
@@ -3,45 +3,47 @@ let { createFile, generateFingerprint } = require("faucet-pipeline-util");
 let { readFile } = require("./promisified-fs");
 let tree = require("./tree");
 
-module.exports = ({ source, target, fingerprint, configDir }) => {
+module.exports = config => {
 	return files => {
 		let filesToCopy;
 
 		if(files) {
-			filesToCopy = filesWithinDirectory(source, files);
+			filesToCopy = filesWithinDirectory(config.source, files);
 		} else {
-			filesToCopy = tree(source);
+			filesToCopy = tree(config.source);
 		}
 
 		return filesToCopy.
 			then(files => files.filter(file => !file.startsWith("."))).
-			then(sources => {
-				return Promise.all(sources.map(fileName => {
-					let sourcePath = path.join(source, fileName);
-
-					return readFile(sourcePath).then(content => {
-						let targetPath = path.join(target, fileName);
-						let outputPath = targetPath;
-
-						if(fingerprint) {
-							outputPath = path.join(target,
-									generateFingerprint(fileName, content));
-						}
-
-						return createFile(outputPath, content).then(_ => {
-							return {
-								changed: true,
-								target: path.relative(configDir, targetPath),
-								output: path.relative(configDir, outputPath),
-								filePath: path.relative(target, outputPath)
-							};
-						});
-					});
-				}));
-			});
+			then(sources => processFiles(sources, config));
 	};
 };
 
+function processFiles(fileNames, config) {
+	return Promise.all(fileNames.map(fileName => processFile(fileName, config)));
+}
+
+function processFile(fileName, { source, target, fingerprint, configDir }) {
+	let sourcePath = path.join(source, fileName);
+	let targetPath = path.join(target, fileName);
+	let outputPath = targetPath;
+
+	return readFile(sourcePath).then(content => {
```

Comment:
I much prefer a line break before `#then` because it makes more salient the operation's asynchronous nature.

That would also have made it obvious that you don't need nested promises below:

```javascript
return readFile(sourcePath).
    then(content => {
        …
        return createFile(outputPath, content);
    }).
    then(_ => ({
        changed: true,
        …
    }))
```

Rule of thumb: Be wary of indentation, more than a couple of levels should raise olfactory suspicion.

Incidentally, that's why this PR is clearly an improvement over the previous code. 👍